packages/ast/src/clients/rpc/class/grpc-gateway/utils/grpc-gateway.ts

```diff
@@ -0,0 +1,158 @@
+            ),
+        )
+    ),
+    true,
+)
+
+// initReqProperties contains information for initReq parameter in fetchReq arguments
+const getInitReqProperties = () => {
+    const initReqProperties = [];
+
+    // <...initReq>
+    const argSpreadInit: t.SpreadElement = t.spreadElement(
+        t.identifier('initReq')
+    )
+
+    // <method: 'POST'>
+    const argPOST: t.ObjectProperty = t.objectProperty(
+        t.identifier('method'),
+        t.stringLiteral('POST'),
+        false,
+        false,
+    )
+
+    // <JSON.stringify(req, fm.replacer)>
+    const argBody: t.ObjectProperty = t.objectProperty(
+        t.identifier('body'),
+        t.callExpression(
+            t.memberExpression(
+                t.identifier('JSON'),
+                t.identifier('stringify'),
+                false,
+            ),
+            [
+                t.identifier('req'),
+                t.memberExpression(
+                    t.identifier('fm'),
+                    t.identifier('replacer'),
+                    false
+                )
+            ]
+        )
+    )
+
+    initReqProperties.push(argSpreadInit, argPOST, argBody)
+    return initReqProperties
+}
+
+// fetchArgs will be used in method body's return statement expression.
+// Contains arguments to fm.fetchReq
+const getFetchReqArgs = (
+    name: string,
+    packageImport: string
+) => {
+    const fetchArgs = [];
+
+    // first argument of fetchReq
+    const argTemplateLiteral = t.templateLiteral(
+        [
+            t.templateElement(
+                {
+                    // todo: make dynamic
+                    raw: '/' + packageImport + '/' + name,
+                    cooked: '/' + packageImport + '/' + name
+                },
+                true,
+            )
+        ], // quasis
+        [], // empty expressions
+    )
+
+    // adds proto path to fetchReq
+    fetchArgs.push(argTemplateLiteral);
+
+    // initReqProperties (contains information for initReq parameter in fetchReq) arguments: 
+    const initReqProperties = getInitReqProperties()
+    
+    const fetchArgsInitReqObj = t.objectExpression(
+        initReqProperties
+    )
+    // adds initReq parameter to fetchReq
+    fetchArgs.push(fetchArgsInitReqObj)
+
+    return fetchArgs
+}
+
+export const grpcGatewayMethodDefinitionNoBody = (
+    context: GenericParseContext,
+    name: string,
+    msg: string,
+    svc: ProtoServiceMethod,
+    packageImport: string
+) => {
+    console.log(packageImport);
+    const requestType = svc.requestType;
+    const responseType = svc.responseType;
+
+    // first parameter in method
+    // ex: static Send(request: MsgSend)
+    // paramRequst is an object representing everything in brackets here
+    const paramRequst = identifier(
+        'request',
+        t.tsTypeAnnotation(
+            t.tsTypeReference(
+                t.identifier(requestType),
+            )
+        ),
+        false // todo: work around optional
```

Comment:
optionality should be similar to https://github.com/osmosis-labs/telescope/blob/db2456e109aa2bbf513c177ccd6cfb689ea49034/packages/ast/src/clients/rpc/class/rpc.ts#L46-L54

Reply:
thanks! I just wanted to make sure I do this thing in the future

Reply:
overall, what is left until completion:

- build gateway query class (it is different from msg so I can't really use the already build one)
- correctly import all `fm` helper

 and I think basically that's it